pinot-connectors/pinot-flink-connector/src/main/java/org/apache/pinot/connector/flink/sink/FlinkSegmentWriter.java

```diff
@@ -124,12 +135,26 @@ public void init(TableConfig tableConfig, Schema schema, Map<String, String> bat
         "batchConfigMaps must contain only 1 BatchConfig for table: %s", _tableNameWithType);
 
     Map<String, String> batchConfigMap = _batchIngestionConfig.getBatchConfigMaps().get(0);
+    batchConfigMap.put(BatchConfigProperties.UPLOADED_REALTIME_PARTITION_ID, Integer.toString(_indexOfSubtask));
+    batchConfigMap.put(BatchConfigProperties.SEGMENT_UPLOAD_TIME_MS, String.valueOf(_segmentUploadTimeMs));
+    batchConfigMap.computeIfAbsent(
+        BatchConfigProperties.SEGMENT_NAME_GENERATOR_PROP_PREFIX + "." + BatchConfigProperties.SEGMENT_NAME_PREFIX,
```

Comment:
Did not understand the behaviour change. `flink` is supposed to be a default prefix as `UploadedRealtimeSegmentName` mandates a prefix, one can use the batch config to modify the default value and handling of `UploadedRealtimeSegmentName` in Pinot is agnostic of the prefix keyword.

Reply:
I believe this writer is shared for both upsert and non-upsert table. It will be behavior change if we add `flink` prefix to non-upsert table

Reply:
Makes sense, mandating UploadedRealtimeSegmentName naming can be only kept for upsert table which didn't work before. Will update.

Though, I do think we can make UploadedRealtimeSegmentName as default naming convention for uploaded segments for non upsert as well as it will lead to a better segment assignment which currently happens by taking mod with 10_000


```
    if (segmentPartitionId == null) {
      // This case is for the uploaded segments for which there's no partition information.
      // A random, but consistent, partition id is calculated based on the hash code of the segment name.
      // Note that '% 10K' is used to prevent having partition ids with large value which will be problematic later in
      // instance assignment formula.
      segmentPartitionId = Math.abs(segmentName.hashCode() % 10_000);
  }
```

Reply:
Refactored to use the naming convention only for upsert table:
https://github.com/apache/pinot/pull/13837/files#diff-72b89e65ea292a808b5d8a902d8a64adb87834534414d5b9d0e46c4725a0455cR156